template/src/components/ui/CustomInputField.tsx

```diff
@@ -1,43 +1,76 @@
-import { TextField, TextFieldProps } from "@mui/material";
-import { FieldInputProps } from "formik";
+import { MenuItem, TextField, TextFieldProps } from "@mui/material";
+import { FieldInputProps, FormikState, getIn } from "formik";
+import * as React from "react";
 import { FieldError } from "./FieldError";
 
 type IProps = TextFieldProps & {
-    errorMessage?: string;
     field: FieldInputProps<string>;
-    isTouched?: boolean;
+    onChange: () => void;
+    form: FormikState<any>;
+    showValidationErrorText?: boolean;
+    selectOptions?: { value: string; label: string }[];
 };
 
 export const CustomInputField = ({
     style,
     label,
     type,
     autoComplete,
+    multiline,
+    minRows,
+    maxRows,
     required,
-    errorMessage,
+    form,
     field,
-    isTouched,
-    ...props
+    "aria-label": ariaLabel,
+    placeholder,
+    showValidationErrorText = true,
+    select,
+    selectOptions,
+    onChange,
 }: IProps) => {
-    const showError = isTouched && !!errorMessage;
+    const fieldError = getIn(form.errors, field.name);
+    const showError = getIn(form.touched, field.name) && !!fieldError;
+
+    const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+        field.onChange(event);
+
+        if (onChange) {
```

Comment:
`onChange` ist nicht optional getyped, das würde den Check obsolet machen. Richtig typen oder Check entfernen.
@gottfired wollen wir in optional cases auf die Schreibweise `onChange?.()` umstellen?

Reply:
Good catch. Ich wusste gar nicht dass `?.` möglich ist.